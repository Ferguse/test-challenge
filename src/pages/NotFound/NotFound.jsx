import { Link } from 'react-router-dom';

export const NotFound = () => {
  return (
    <div className="bg-white content">
      <div className="container">
        <h2>Page not found</h2>
        <p>
          <Link to="/">Go to the home page</Link>
        </p>
      </div>
    </div>
  );
};
