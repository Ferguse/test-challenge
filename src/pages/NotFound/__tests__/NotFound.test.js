import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockedStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { NotFound } from '../NotFound';

describe('Page [NotFound]', () => {
  const state = {
    wishList: {},
    categories: {},
    current: {},
  };

  const store = configureMockedStore([thunk])(state);
  const history = createMemoryHistory({ initialEntries: ['/'] });

  it('Renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history} location={history.location} navigator={history}>
            <NotFound />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
