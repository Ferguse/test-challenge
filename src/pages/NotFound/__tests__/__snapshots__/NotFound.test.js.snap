// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page [NotFound] Renders 1`] = `
<div
  className="bg-white content"
>
  <div
    className="container"
  >
    <h2>
      Page not found
    </h2>
    <p>
      <a
        href="/"
        onClick={[Function]}
      >
        Go to the home page
      </a>
    </p>
  </div>
</div>
`;
